rct229/rulesets/ashrae9012019/section19/section19rule10.py

```diff
@@ -0,0 +1,225 @@
+
+NOT_APPLICABLE_CLIMATE_ZONE = ["CZ0A", "CZ0B", "CZ1A", "CZ1B", "CZ2A", "CZ3A", "CZ4A"]
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_3,
+    HVAC_SYS.SYS_4,
+    HVAC_SYS.SYS_5,
+    HVAC_SYS.SYS_6,
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_8,
+    HVAC_SYS.SYS_11,
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_13,
+]
+
+SYSTEM_3_4_TYPES = [HVAC_SYS.SYS_3, HVAC_SYS.SYS_4]
+
+AIR_ECONOMIZER = schema_enums["AirEconomizerOptions"]
+LIGHTING_BUILDING_AREA = schema_enums[
+    "LightingBuildingAreaOptions2019ASHRAE901T951TG38"
+]
+
+
+class Section19Rule10(RuleDefinitionListIndexedBase):
+    """Rule 10 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule10, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section19Rule10.HVACRule(),
+            index_rmr="baseline",
+            id="19-10",
+            description="Air economizers shall be included in baseline HVAC Systems 3 through 8, and 11, 12, and 13 based on climate as specified in Section G3.1.2.6 with exceptions."
+            "1. Systems that include gas-phase air cleaning to meet the requirements of Standard 62.1, Section 6.1.2. This exception shall be used only if the system in the proposed design does not match the building design."
+            "2. Where the use of outdoor air for cooling will affect supermarket open refrigerated case-work systems. This exception shall only be used if the system in the proposed design does not use an economizer. If the exception is used, an economizer shall not be included in the baseline building design."
+            "3. Systems that serve computer rooms complying with Section G3.1.2.6.1.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.6 including exceptions 1-3",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        proposed_has_economizer = is_economizer_modeled_in_proposed(rmi_b, rmi_p)
+
+        hvac_system_exception_2_list = []
+        if rmi_b.get(
+            "refrigerated_cases"
+        ) is not None and not is_economizer_modeled_in_proposed(rmi_b, rmi_p):
+            hvac_system_exception_2_list = [
+                hvac_id_b
+                for hvac_id_b in find_all(
+                    f'$.buildings[*].building_segments[?(@.lighting_building_area_type == "{LIGHTING_BUILDING_AREA.RETAIL}")].heating_ventilating_air_conditioning_systems[*].id',
```

Comment:
What if a retail project is using the space-by-space method to set their lighting power densities? Will the LIGHTING_BUILDING_AREA property show up in this case?

Reply:
i believe we don't need to think about the space-by-space method case because [This line](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section19/Rule19-10.md#:~:text=Check%20if%20the%20building%20area%20type%20is%20retail%3A%20if%20bldg_seg.lighting_building_area_type%20%3D%3D%20%22RETAIL%22%3A) in the RDS checks the ```RETAIL``` case.